lib/pbench/test/unit/server/test_cache_manager.py

```diff
@@ -503,57 +506,80 @@ def generate_test_result_tree(tmp_path: Path, dir_name: str) -> Path:
     class MockTarball:
         def __init__(self, path: Path, controller: Controller):
             self.name = Dataset.stem(path)
+            self.resource_id = "ABC"
             self.tarball_path = path
-            self.cache = controller.cache / "ABC"
+            self.cache = controller.cache / self.resource_id
+            self.lock = self.cache / "lock"
+            self.last_ref = self.cache / "last_ref"
             self.unpacked = None
+            self.controller = controller
 
-    def test_unpack_tar_subprocess_exception(self, make_logger, monkeypatch):
+    def test_unpack_tar_subprocess_exception(
+        self, make_logger, db_session, monkeypatch
+    ):
         """Show that, when unpacking of the Tarball fails and raises
         an Exception it is handled successfully."""
         tar = Path("/mock/A.tar.xz")
         cache = Path("/mock/.cache")
-        rmtree_called = False
 
-        def mock_rmtree(path: Path, ignore_errors=False):
-            nonlocal rmtree_called
-            rmtree_called = True
+        locks: list[tuple[str, str]] = []
```

Comment:
Yeah, I collected this and cloned it with the mocks, but only a few test cases actually validate it. I chose not to worry about that too much, though I'm sure you'll object. 😆